File: services/assessment-service/src/main/java/com/visor/school/assessment/repository/ExamResultCollectionRepository.java
package com.visor.school.assessment.repository;

import com.visor.school.assessment.model.ExamResultCollection;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface ExamResultCollectionRepository extends JpaRepository<ExamResultCollection, UUID> {
    List<ExamResultCollection> findByClassIdAndAcademicYearAndTerm(
        UUID classId,
        String academicYear,
        String term
    );

    List<ExamResultCollection> findByCollectedBy(UUID collectedBy);
}

File: services/academic-assessment-service/src/main/kotlin/com/visor/school/assessment/repository/ExamResultCollectionRepository.kt
package com.visor.school.assessment.repository

import com.visor.school.assessment.model.ExamResultCollection
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface ExamResultCollectionRepository : JpaRepository<ExamResultCollection, UUID> {
    fun findByClassIdAndAcademicYearAndTerm(
        classId: UUID,
        academicYear: String,
        term: String
    ): List<ExamResultCollection>

    fun findByCollectedBy(collectedBy: UUID): List<ExamResultCollection>
}

